packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -1202,13 +1236,26 @@ class ShortcutRegistry with ChangeNotifier {
   ///    shortcuts associated with a particular entry.
   ShortcutRegistryEntry addAll(Map<ShortcutActivator, Intent> value) {
     assert(ChangeNotifier.debugAssertNotDisposed(this));
+    assert(value.isNotEmpty, 'Cannot register an empty map of shortcuts');
     final ShortcutRegistryEntry entry = ShortcutRegistryEntry._(this);
-    _tokenShortcuts[entry] = value;
+    _registeredShortcuts[entry] = value;
     assert(_debugCheckForDuplicates());
-    notifyListeners();
+    _notifyListenersNextFrame();
     return entry;
   }
 
+  void _notifyListenersNextFrame() {
```

Comment:
Can you explain the motivation behind _notifyListenersNextFrame and this change to ShortcutRegistry?